tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1338,38 +1357,385 @@ static void test_rx_priv_interleave(void)
+
+struct priv_test_ctx {
+	uint8_t beacon_type;
+	uint16_t *node_id_addr;
+};
+
+static void pp_netkey_ctx_init(struct pp_netkey_ctx *net)
+{
+	int err;
+
+	err = bt_mesh_identity_key(net->net_key, &net->id_key);
+	if (err) {
+		FAIL("Failed to generate ID key (err %d)", err);
+	}
+
+	err = bt_mesh_k3(net->net_key, net->net_id);
+	if (err) {
+		FAIL("Failed to generate Net ID (err %d)", err);
+	}
+}
+
+static bool pp_type_check(uint16_t expected_beacon, uint8_t adv_type, struct net_buf_simple *buf)
+{
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND || buf->len != BEACON_TYPE_PRIVATE_LEN) {
+		return false;
+	}
+
+	/* Remove Header */
+	(void)net_buf_simple_pull_mem(buf, 11);
+
+	uint8_t beacon_type = net_buf_simple_pull_u8(buf);
+
+	if (beacon_type != expected_beacon) {
+		return false;
+	}
+
+	return true;
+}
+
+static uint64_t pp_hash_calc(struct pp_netkey_ctx *net, uint64_t random, uint16_t *addr)
+{
+	uint64_t hash;
+	uint8_t tmp[16] = {0, 0, 0, 0, 0, 3};
+
+	if (addr) {
+		memcpy(&tmp[6], &random, 8);
+		sys_put_be16(*addr, &tmp[14]);
+
+	} else {
+		memcpy(&tmp[0], net->net_id, 8);
+		memcpy(&tmp[8], &random, 8);
+	}
+
+	bt_mesh_encrypt(&net->id_key, tmp, tmp);
+	memcpy(&hash, &tmp[8], 8);
+
+	return hash;
+}
+
+static bool pp_beacon_check(const uint8_t *net_id, void *ctx)
+{
+	struct priv_test_ctx *test_ctx = (struct priv_test_ctx *)ctx;
+
+	ASSERT_EQUAL(beacon.pp_hash,
+		     pp_hash_calc(&pp_net0, beacon.pp_random, test_ctx->node_id_addr));
+
+	if (memcmp(beacon.adv_addr.a.val, last_beacon_adv_addr.a.val, BT_ADDR_SIZE) == 0) {
+		return false;
+	}
+
+	memcpy(&last_beacon_adv_addr.a.val, beacon.adv_addr.a.val, BT_ADDR_SIZE);
+
+	return true;
+}
+
+static void priv_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				struct net_buf_simple *buf)
+{
+	struct priv_test_ctx *ctx = (struct priv_test_ctx *)beacon.user_ctx;
+
+	if (!pp_type_check(ctx->beacon_type, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	bt_addr_le_copy(&beacon.adv_addr, addr);
+
+	beacon.pp_hash = net_buf_simple_pull_le64(buf);
+	beacon.pp_random = net_buf_simple_pull_le64(buf);
+
+	if (!beacon.process_cb || beacon.process_cb(NULL, beacon.user_ctx)) {
+		k_sem_give(&observer_sem);
+	}
+}
+
+static void rx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(&rx_cfg, wait);
+	bt_mesh_crypto_init();
+	pp_netkey_ctx_init(&pp_net0);
+	k_sem_init(&observer_sem, 0, 1);
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+		return;
+	}
+}
+
+static void tx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(NULL, wait);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+
+	/* Disable GATT proxy */
+	err = bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED);
+	if (err) {
+		FAIL("Failed to disable gatt proxy (err %d)", err);
+	}
+}
+
+static void test_tx_priv_net_id(void)
+{
+	int err;
+
+	tx_priv_common_init(610);
```

Comment:
Please use macros for initialization delays. If tests stop working due to performance issue, it will be nice to have changes in place of macros declaration rather than everywhere in the code.